### PR TITLE
Fix convert_advanced_parameters_to_dict

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Replaceconvert_advanced_parameters_to_dict to apply_advanced_parameters_to_config to avoid missing parameters on config.update().

```python
a = {"a": 1, "b": {"b1": 1, "b2": 2}} 
b = {"b": {"b1": "b1"}} 
a.update(b)  # {"a": 1, "b": {"b1": "b1"}},  missed "b2": 2
```
As result of dict.update batchnorm_adaptation is missed after update config with result of convert_advanced_parameters_to_dict.

### Reason for changes

Incorrect converting advanced parameters to NNCFConfig.

### Tests

test_create_nncf_config

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
run tensorflow pre-commit tests

**Comment by AlexanderDokuchaev** [comment]
run tensorflow pre-commit tests

File: nncf/tensorflow/quantization/quantize_model.py [hunk 1]
```diff
@@ -124,16 +124,7 @@ def _create_nncf_config(
...
 
     if advanced_parameters is not None:
-        advanced_config = convert_advanced_parameters_to_dict(advanced_parameters)
-
-        ranges = advanced_config.get("initializer", {}).get("range")
-        if ranges is not None:
-            for rconfig in ranges:
-                rconfig["num_init_samples"] = subset_size
-                if "type" not in rconfig:
-                    rconfig["type"] = DEFAULT_RANGE_TYPE
```
Comment (alexsu52) [inline nncf/tensorflow/quantization/quantize_model.py:134]: it looks like you should set default parameters for "type" and "num_init_samples" in activation and weights ranges

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: nncf/tensorflow/quantization/quantize_model.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply nncf/tensorflow/quantization/quantize_model.py:134]: The `type` field set in `_get_default_quantization_config`  https://github.com/openvinotoolkit/nncf/blob/67a0be8b3b3848a6ba3bf23ee15f0df26e83050e/nncf/torch/quantization/quantize_model.py#L125

On this line types set again because `config.update` overrites `config["initializer"]`. I describe it in description of this PR.

Comment (AlexanderDokuchaev) [reply nncf/tensorflow/quantization/quantize_model.py:134]: Fill `type` and `num_init_samples` in `apply_advanced_parameters_to_config` from part of config that generated by `_get_default_quantization_config`: https://github.com/openvinotoolkit/nncf/pull/1906/files#diff-950f24ab49d1eb3db259be96703abb72ea21125416cdf9b110357754a3bfaa8fR333-R336 

And add test to test this case.

**Comment by AlexanderDokuchaev** [comment]
run pytorcch pre-commit tests

**Review by alexsu52 — APPROVED** [review]
LGTM
